### CS_5661 PROJECT : RedHat Business Prediction

#### Step 1: Import all the Libraries needed

In [1]:
import pandas as pd
import numpy as np
import xlrd

#### Step 2 : Read the people dataset & Print it

In [2]:
df = pd.read_csv("C:/Masters/5661/Project REDHAT/people/people.csv")
df.head()


OSError: File b'C:/Masters/5661/Project REDHAT/people/people.csv' does not exist

#### Step 3 : Read the activity_training dataset & print it

In [96]:
df_train1 = pd.read_csv("C:/Masters/5661/Project REDHAT/act_train/act_train.csv")
#df_train2 = pd.read_csv("C:/Masters/5661/Project REDHAT/act_train/act_train.csv")
df_train1.head()

people_id   activity_id        date activity_category char_1 char_2 char_3  \
0   ppl_100  act2_1734928  2023-08-26            type 4    NaN    NaN    NaN   
1   ppl_100  act2_2434093  2022-09-27            type 2    NaN    NaN    NaN   
2   ppl_100  act2_3404049  2022-09-27            type 2    NaN    NaN    NaN   
3   ppl_100  act2_3651215  2023-08-04            type 2    NaN    NaN    NaN   
4   ppl_100  act2_4109017  2023-08-26            type 2    NaN    NaN    NaN   

  char_4 char_5 char_6 char_7 char_8 char_9  char_10  outcome  
0    NaN    NaN    NaN    NaN    NaN    NaN  type 76        0  
1    NaN    NaN    NaN    NaN    NaN    NaN   type 1        0  
2    NaN    NaN    NaN    NaN    NaN    NaN   type 1        0  
3    NaN    NaN    NaN    NaN    NaN    NaN   type 1        0  
4    NaN    NaN    NaN    NaN    NaN    NaN   type 1        0

#### Step 4 : Drop some features which are difficult to convert in Activity_training Dataset

In [97]:
#Drop Charector column that are not required from feature matrix in activity table
df_train1 = df_train1.drop('activity_id', axis=1)
df_train1 = df_train1.drop('date', axis=1)
df_train1 = df_train1.drop('char_10', axis=1)

# drop colums with NAN for easier machine learning and remove complexity
df_train1 = df_train1.dropna(how='any')

print df_train1.head()
df_train1.shape

      people_id activity_category   char_1   char_2  char_3  char_4  char_5  \
52   ppl_100025            type 1   type 3   type 5  type 1  type 1  type 6   
105  ppl_100033            type 1  type 36  type 11  type 5  type 1  type 6   
106  ppl_100033            type 1  type 24   type 6  type 6  type 3  type 1   
107  ppl_100033            type 1   type 2   type 2  type 3  type 3  type 5   
108  ppl_100033            type 1   type 2   type 5  type 3  type 2  type 6   

     char_6  char_7  char_8  char_9  outcome  
52   type 3  type 3  type 6  type 8        0  
105  type 1  type 1  type 4  type 1        0  
106  type 3  type 4  type 5  type 1        0  
107  type 2  type 2  type 4  type 2        0  
108  type 1  type 1  type 6  type 8        0  


(157615, 12)

#### Step 5: Perform One Hot Encoding on Activity_training Dataset

In [98]:
# perform Onehot encoding on remaining feature colums
cols_to_transform = ['activity_category','char_1','char_2','char_3','char_4','char_5','char_6','char_7','char_8','char_9']
for colums in cols_to_transform:
    df_with_dummies = pd.get_dummies(df_train1[colums])
    df_train1 = df_train1.drop(colums, axis=1)
    df_train1 = df_train1.join(df_with_dummies, lsuffix='_left', rsuffix='_right')

print df_train1.head()
df_train1.shape

      people_id  outcome  type 1_left  type 1_right  type 10_left  \
52   ppl_100025        0          1.0           0.0           0.0   
105  ppl_100033        0          1.0           0.0           0.0   
106  ppl_100033        0          1.0           0.0           0.0   
107  ppl_100033        0          1.0           0.0           0.0   
108  ppl_100033        0          1.0           0.0           0.0   

     type 11_left  type 12_left  type 13_left  type 14_left  type 15_left  \
52            0.0           0.0           0.0           0.0           0.0   
105           0.0           0.0           0.0           0.0           0.0   
106           0.0           0.0           0.0           0.0           0.0   
107           0.0           0.0           0.0           0.0           0.0   
108           0.0           0.0           0.0           0.0           0.0   

      ...    type 18_right  type 19  type 2  type 3  type 4  type 5  \
52    ...              0.0      0.0     0.0     0.0

(157615, 161)

#### Step 6 : Define Label Vector

In [99]:
y_activity=df_train1['outcome']
y_activity

52         0
105        0
106        0
107        0
108        0
124        1
125        1
126        1
127        1
128        1
129        1
130        1
186        0
187        0
206        0
219        1
220        1
231        0
232        0
243        0
250        1
251        1
252        1
253        1
254        1
261        0
262        1
263        0
268        0
279        1
          ..
2196920    0
2196922    0
2196927    0
2196929    1
2196930    1
2196931    1
2197021    0
2197022    0
2197023    0
2197024    0
2197025    0
2197026    0
2197027    0
2197028    0
2197029    0
2197030    0
2197031    0
2197032    0
2197035    1
2197144    0
2197157    0
2197158    0
2197163    0
2197192    0
2197195    0
2197196    0
2197220    1
2197227    1
2197233    0
2197245    1
Name: outcome, dtype: int64

In [100]:
#drop label column that are not required from feature matrix in activity table to seperate y axis
df_train1 = df_train1.drop('outcome', axis=1)
df_train1.head()

people_id  type 1_left  type 1_right  type 10_left  type 11_left  \
52   ppl_100025          1.0           0.0           0.0           0.0   
105  ppl_100033          1.0           0.0           0.0           0.0   
106  ppl_100033          1.0           0.0           0.0           0.0   
107  ppl_100033          1.0           0.0           0.0           0.0   
108  ppl_100033          1.0           0.0           0.0           0.0   

     type 12_left  type 13_left  type 14_left  type 15_left  type 16_left  \
52            0.0           0.0           0.0           0.0           0.0   
105           0.0           0.0           0.0           0.0           0.0   
106           0.0           0.0           0.0           0.0           0.0   
107           0.0           0.0           0.0           0.0           0.0   
108           0.0           0.0           0.0           0.0           0.0   

      ...    type 18_right  type 19  type 2  type 3  type 4  type 5  \
52    ...              0.0      0.0     0.0     0.0     0.0     0.0   
105   ...              0.0      0.0     0.0     0.0     0.0     0.0   
106   ...              0.0      0.0     0.0     0.0     0.0     0.0   
107   ...              0.0      0.0     1.0     0.0     0.0     0.0   
108   ...              0.0      0.0     0.0     0.0     0.0     0.0   

     type 6_right  type 7_right  type 8_right  type 9  
52            0.0           0.0           1.0     0.0  
105           0.0           0.0           0.0     0.0  
106           0.0           0.0           0.0     0.0  
107           0.0           0.0           0.0     0.0  
108           0.0           0.0           1.0     0.0  

[5 rows x 160 columns]

#### Step 6 : Drop some features which are difficult to convert in People Dataset

In [101]:
#drop feature column that are not required from feature matrix in people table 
df = df.drop('group_1', axis=1)
df = df.drop('date', axis=1)

print df.head()
df.shape

    people_id  char_1  char_2   char_3   char_4  char_5  char_6   char_7  \
0     ppl_100  type 2  type 2   type 5   type 5  type 5  type 3  type 11   
1  ppl_100002  type 2  type 3  type 28   type 9  type 5  type 3  type 11   
2  ppl_100003  type 2  type 3   type 4   type 8  type 5  type 2   type 5   
3  ppl_100004  type 2  type 3  type 40  type 25  type 9  type 4  type 16   
4  ppl_100006  type 2  type 3  type 40  type 25  type 9  type 3   type 8   

   char_8  char_9   ...   char_29 char_30 char_31 char_32 char_33 char_34  \
0  type 2  type 2   ...     False    True    True   False   False    True   
1  type 2  type 4   ...     False    True    True    True    True    True   
2  type 2  type 2   ...     False   False    True    True    True    True   
3  type 2  type 2   ...      True    True    True    True    True    True   
4  type 2  type 2   ...     False   False    True   False   False   False   

  char_35 char_36 char_37 char_38  
0    True    True   False      36  
1    Tru

(189118, 39)

#### Step 7: Perform One Hot Encoding on People Dataset

In [102]:
# perform Onehot encoding on remaining feature colums in people table
col_trans_ppl = ['char_1','char_2','char_3','char_4','char_5','char_6','char_7','char_8','char_9']
for colums in col_trans_ppl:
    df_dummies = pd.get_dummies(df[colums])
    df = df.drop(colums, axis=1)
    df = df.join(df_dummies, lsuffix='_left', rsuffix='_right')

print df.head()
df.shape

    people_id char_10 char_11 char_12 char_13 char_14 char_15 char_16 char_17  \
0     ppl_100    True   False   False    True    True   False    True   False   
1  ppl_100002   False   False    True    True   False   False   False    True   
2  ppl_100003    True    True    True    True    True    True   False    True   
3  ppl_100004    True    True    True    True    True   False    True    True   
4  ppl_100006   False   False   False   False   False   False   False   False   

  char_18  ...   type 8_left type 1 type 2 type 3_right type 4 type 5 type 6  \
0   False  ...           0.0    0.0    1.0          0.0    0.0    0.0    0.0   
1   False  ...           0.0    0.0    0.0          0.0    1.0    0.0    0.0   
2   False  ...           0.0    0.0    1.0          0.0    0.0    0.0    0.0   
3    True  ...           0.0    0.0    1.0          0.0    0.0    0.0    0.0   
4   False  ...           0.0    0.0    1.0          0.0    0.0    0.0    0.0   

  type 7 type 8_right type 9  
0

(189118, 161)

#### Step 8 : Merge the two Datasets

In [103]:
df_new = df_train1.merge(df, on='people_id',how='left')
df_new.head()

people_id  type 1_left_x  type 1_right_x  type 10_left_x  type 11_left_x  \
0  ppl_100025            1.0             0.0             0.0             0.0   
1  ppl_100033            1.0             0.0             0.0             0.0   
2  ppl_100033            1.0             0.0             0.0             0.0   
3  ppl_100033            1.0             0.0             0.0             0.0   
4  ppl_100033            1.0             0.0             0.0             0.0   

   type 12_left_x  type 13_left_x  type 14_left_x  type 15_left_x  \
0             0.0             0.0             0.0             0.0   
1             0.0             0.0             0.0             0.0   
2             0.0             0.0             0.0             0.0   
3             0.0             0.0             0.0             0.0   
4             0.0             0.0             0.0             0.0   

   type 16_left_x    ...     type 8_left_y  type 1  type 2_y  type 3_right_y  \
0             0.0    ...               0.0     0.0       0.0             0.0   
1             0.0    ...               0.0     0.0       0.0             1.0   
2             0.0    ...               0.0     0.0       0.0             1.0   
3             0.0    ...               0.0     0.0       0.0             1.0   
4             0.0    ...               0.0     0.0       0.0             1.0   

   type 4_y  type 5_y  type 6  type 7  type 8_right_y  type 9_y  
0       0.0       0.0     1.0     0.0             0.0       0.0  
1       0.0       0.0     0.0     0.0             0.0       0.0  
2       0.0       0.0     0.0     0.0             0.0       0.0  
3       0.0       0.0     0.0     0.0             0.0       0.0  
4       0.0       0.0     0.0     0.0             0.0       0.0  

[5 rows x 320 columns]

In [104]:


#df_new['people_id'] = df_new['people_id'].astype('float64') 
#df_new = df_new.convert_objects(convert_numeric=True)
#df_new['people_id'] = pd.to_numeric(df_new['people_id'])

#Drop the People id as we have a updated Merged table with people and activity
df_new = df_new.drop('people_id', axis=1)
X = df_new

Y = y_activity

#### Step 9 : Split the data to do training and check for testing

In [149]:
# Randomly splitting the original dataset into training set and testing set
# The function"train_test_split" from "sklearn.cross_validation" library performs random splitting.
# "test_size=0.3" means that pick 30% of data samples for testing set, and the rest (70%) for training set.

from sklearn.cross_validation import train_test_split
#from sklearn.model_selection import train_test_split

x_train, x_test, y_train, y_test = train_test_split(X, Y, test_size=0.3, random_state=2)

In [150]:
print x_train.shape
print x_test.shape
print y_train.shape
print y_test.shape

(110330, 319)
(47285, 319)
(110330L,)
(47285L,)


#### Step 10 : Reduce Dimensions as the features are too large

In [105]:
### Import library for PCA
from sklearn.decomposition import PCA

In [151]:
# reduce dimensions. Use PCA
pca = PCA(n_components=50)
# train the algorithim
pca.fit(x_train)
# transform the vector for training set
X_transform = pca.fit_transform(x_train)

# put it into a dataframe
X_transform = pd.DataFrame(X_transform)
print X_transform.head()

          0         1         2         3         4         5         6   \
0 -29.234324 -0.518611 -0.800467 -0.528117  1.504308  0.154077  0.102118   
1  34.862525  1.969071 -0.567406 -0.030956  0.051764 -0.071096 -0.679222   
2  33.863945  2.208835  1.421296 -0.573907  0.378862 -0.489495 -0.643617   
3  40.905938  2.709339  1.087537 -1.154597  0.386430 -0.504103  2.343148   
4  36.757652 -2.100111  1.463354 -1.015044  0.628493 -0.919552  0.767708   

         7         8         9     ...           40        41        42  \
0  0.631115  0.776177 -1.059978    ...     0.074444  0.038706 -0.274787   
1 -0.251680 -0.016075  0.468662    ...    -0.593657  0.330028  0.342384   
2 -0.538944  0.035243  0.943296    ...     0.384120  0.102347 -0.189715   
3  0.199373  0.229629  0.419053    ...    -0.046309  0.294651  0.308221   
4 -0.469864 -0.612968 -0.523451    ...     0.333443 -0.170863  0.004188   

         43        44        45        46        47        48        49  
0 -0.046073 -0.100

In [136]:
# transform the vector for testing set
X_transform_test = pca.fit_transform(x_test)
# put it into a dataframe
X_transform_test = pd.DataFrame(X_transform_test)

X_transform_test.head()

0         1         2         3         4         5         6   \
0  33.041476 -2.142130  1.227109 -0.951787  0.887987 -0.459223  1.004788   
1  41.158097  1.545542  0.765306  0.502839 -1.194719 -0.924412 -0.718672   
2  31.050672 -2.075585 -1.510147  0.711042  0.838020 -0.910129 -0.747553   
3  37.173220  1.982169  0.519908 -0.619307  1.262522  0.090931 -0.359381   
4 -48.954205 -0.040434 -0.961269  1.633794  0.896826 -0.315696  0.390336   

         7         8         9     ...           40        41        42  \
0  0.153057 -0.463672 -0.090824    ...     0.454618  0.413261 -0.171351   
1  0.992360  0.598896 -0.860971    ...    -0.494734 -0.221816  0.006538   
2 -0.466721  0.113831 -0.281059    ...     0.066322  0.178402  0.012317   
3  0.547415  0.485518 -0.308729    ...     0.756567  0.394234  0.337450   
4 -0.082967 -0.398101 -0.105660    ...    -0.206368  0.160852  0.239504   

         43        44        45        46        47        48        49  
0 -0.476556 -0.033931  0.307234  0.248188  0.219667 -0.106704 -0.409097  
1  0.797897 -0.434740  0.185805 -0.217531  0.196225 -0.328749  0.029738  
2 -0.208403  0.340111 -0.194930  0.294749 -0.014473 -0.008838 -0.299654  
3 -0.004689 -0.693237 -0.272591  0.162378 -0.445989  0.332378  0.020884  
4  0.159363 -0.248012 -0.066552 -0.082041  0.043514 -0.016597 -0.205299  

[5 rows x 50 columns]

#### Step 11 : Perform random Forest

In [145]:
#Random Forest is an classifier using multiple decision trees to arive a best prediction

from sklearn.ensemble import RandomForestClassifier

my_RandomForest = RandomForestClassifier(n_estimators = 19, bootstrap = True,random_state=2)

In [146]:
#train the RandomForest alg by using the FIT method

my_RandomForest.fit(X_transform,y_train,sample_weight=None)

RandomForestClassifier(bootstrap=True, class_weight=None, criterion='gini',
            max_depth=None, max_features='auto', max_leaf_nodes=None,
            min_impurity_split=1e-07, min_samples_leaf=1,
            min_samples_split=2, min_weight_fraction_leaf=0.0,
            n_estimators=25, n_jobs=1, oob_score=False, random_state=2,
            verbose=0, warm_start=False)

In [147]:
#Make prediction over the trained RandomForest alg using the testing sample:

y_RandomForest_predict = my_RandomForest.predict(X_transform_test)

print(y_RandomForest_predict)
print(y_RandomForest_predict.shape)

[1 1 1 ..., 1 0 0]
(31523L,)


In [148]:
# Function "accuracy_score" from "sklearn.metrics" will perform the element-to-element comparision and returns the 
# portion of correct predictions:
from sklearn.metrics import accuracy_score
score_RF = accuracy_score(y_test, y_RandomForest_predict)
print(score_RF)

0.833994226438


#### Accuracy is 0.833994226438

#### Step 12 : Perform AdaBoost

In [115]:
# Adaboost is a classifier which gives weightage to each features and reduces the errors in getting the best prediction

from sklearn.ensemble import AdaBoostClassifier

my_AdaBoost = AdaBoostClassifier(n_estimators=19,random_state=2)

In [116]:
#train the adaboost alg by using the FIT method

my_AdaBoost.fit(X_transform,y_train,sample_weight=None)

AdaBoostClassifier(algorithm='SAMME.R', base_estimator=None,
          learning_rate=1.0, n_estimators=19, random_state=2)

In [117]:
#Make prediction over the trained Adaboost alg using the testing sample:

y_adaboost_predict = my_AdaBoost.predict(X_transform_test)

print(y_adaboost_predict)
print(y_adaboost_predict.shape)

[1 1 1 ..., 1 0 1]
(47285L,)


In [118]:
adaboost_score = accuracy_score(y_test, y_adaboost_predict)

print(adaboost_score)

0.839462831765


#### Accuracy is 0.839462831765

###  TESTING DATASET

#### Step 1 : Read the activity_testing dataset

In [119]:
df_test = pd.read_csv("C:/Masters/5661/Project REDHAT/act_test.csv")

df_test.head()

people_id   activity_id        date activity_category   char_1   char_2  \
0  ppl_100004   act1_249281  2022-07-20            type 1   type 5  type 10   
1  ppl_100004   act2_230855  2022-07-20            type 5      NaN      NaN   
2   ppl_10001   act1_240724  2022-10-14            type 1  type 12   type 1   
3   ppl_10001    act1_83552  2022-11-27            type 1  type 20  type 10   
4   ppl_10001  act2_1043301  2022-10-15            type 5      NaN      NaN   

   char_3  char_4  char_5  char_6  char_7   char_8   char_9    char_10  
0  type 5  type 1  type 6  type 1  type 1   type 7   type 4        NaN  
1     NaN     NaN     NaN     NaN     NaN      NaN      NaN   type 682  
2  type 5  type 4  type 6  type 1  type 1  type 13  type 10        NaN  
3  type 5  type 4  type 6  type 1  type 1   type 5   type 5        NaN  
4     NaN     NaN     NaN     NaN     NaN      NaN      NaN  type 3015

#### Step 2 : Drop the features not required in activity_testing dataset

In [120]:
#drop Charector column that are not required from feature matrix in activity table
df_test = df_test.drop('activity_id', axis=1)
df_test = df_test.drop('date', axis=1)
df_test = df_test.drop('char_10', axis=1)

# drop colums with NAN for easier machine learning and remove complexity
df_test = df_test.dropna(how='any')

print df_test.head()
df_test.shape

     people_id activity_category   char_1   char_2  char_3  char_4  char_5  \
0   ppl_100004            type 1   type 5  type 10  type 5  type 1  type 6   
2    ppl_10001            type 1  type 12   type 1  type 5  type 4  type 6   
3    ppl_10001            type 1  type 20  type 10  type 5  type 4  type 6   
27  ppl_100010            type 1   type 2   type 1  type 9  type 3  type 1   
28  ppl_100010            type 1   type 2   type 1  type 5  type 3  type 1   

    char_6  char_7   char_8   char_9  
0   type 1  type 1   type 7   type 4  
2   type 1  type 1  type 13  type 10  
3   type 1  type 1   type 5   type 5  
27  type 3  type 2   type 2   type 1  
28  type 3  type 1   type 2   type 1  


(40092, 11)

#### Step 3 : Perform One Hot Encoding

In [121]:
# perform Onehot encoding on remaining feature colums with testing dataset
cols_to_transform = ['activity_category','char_1','char_2','char_3','char_4','char_5','char_6','char_7','char_8','char_9']
for colums in cols_to_transform:
    df_test_dummies = pd.get_dummies(df_test[colums])
    df_test = df_test.drop(colums, axis=1)
    df_test = df_test.join(df_test_dummies, lsuffix='_left', rsuffix='_right')

print df_test.head()
df_test.shape

     people_id  type 1_left  type 1_right  type 10_left  type 11_left  \
0   ppl_100004          1.0           0.0           0.0           0.0   
2    ppl_10001          1.0           0.0           0.0           0.0   
3    ppl_10001          1.0           0.0           0.0           0.0   
27  ppl_100010          1.0           0.0           0.0           0.0   
28  ppl_100010          1.0           0.0           0.0           0.0   

    type 12_left  type 13_left  type 14_left  type 15_left  type 16_left  \
0            0.0           0.0           0.0           0.0           0.0   
2            1.0           0.0           0.0           0.0           0.0   
3            0.0           0.0           0.0           0.0           0.0   
27           0.0           0.0           0.0           0.0           0.0   
28           0.0           0.0           0.0           0.0           0.0   

     ...    type 18_right  type 19  type 2  type 3  type 4  type 5  \
0    ...              0.0      0.0

(40092, 155)

#### Step 4 : Merge People dataset with activity_testing datset

In [122]:
df_new_test = df_test.merge(df, on='people_id',how='left')
df_new_test.head()

people_id  type 1_left_x  type 1_right_x  type 10_left_x  type 11_left_x  \
0  ppl_100004            1.0             0.0             0.0             0.0   
1   ppl_10001            1.0             0.0             0.0             0.0   
2   ppl_10001            1.0             0.0             0.0             0.0   
3  ppl_100010            1.0             0.0             0.0             0.0   
4  ppl_100010            1.0             0.0             0.0             0.0   

   type 12_left_x  type 13_left_x  type 14_left_x  type 15_left_x  \
0             0.0             0.0             0.0             0.0   
1             1.0             0.0             0.0             0.0   
2             0.0             0.0             0.0             0.0   
3             0.0             0.0             0.0             0.0   
4             0.0             0.0             0.0             0.0   

   type 16_left_x    ...     type 8_left_y  type 1  type 2_y  type 3_right_y  \
0             0.0    ...               0.0     0.0       1.0             0.0   
1             0.0    ...               0.0     0.0       1.0             0.0   
2             0.0    ...               0.0     0.0       1.0             0.0   
3             0.0    ...               0.0     1.0       0.0             0.0   
4             0.0    ...               0.0     1.0       0.0             0.0   

   type 4_y  type 5_y  type 6  type 7_y  type 8_right_y  type 9_y  
0       0.0       0.0     0.0       0.0             0.0       0.0  
1       0.0       0.0     0.0       0.0             0.0       0.0  
2       0.0       0.0     0.0       0.0             0.0       0.0  
3       0.0       0.0     0.0       0.0             0.0       0.0  
4       0.0       0.0     0.0       0.0             0.0       0.0  

[5 rows x 315 columns]

In [123]:
#Drop the People id as we have a updated Merged table with people and activity
df_new_test = df_new_test.drop('people_id', axis=1)
X_test = df_new_test
X_test.head()


type 1_left_x  type 1_right_x  type 10_left_x  type 11_left_x  \
0            1.0             0.0             0.0             0.0   
1            1.0             0.0             0.0             0.0   
2            1.0             0.0             0.0             0.0   
3            1.0             0.0             0.0             0.0   
4            1.0             0.0             0.0             0.0   

   type 12_left_x  type 13_left_x  type 14_left_x  type 15_left_x  \
0             0.0             0.0             0.0             0.0   
1             1.0             0.0             0.0             0.0   
2             0.0             0.0             0.0             0.0   
3             0.0             0.0             0.0             0.0   
4             0.0             0.0             0.0             0.0   

   type 16_left_x  type 17_left_x    ...     type 8_left_y  type 1  type 2_y  \
0             0.0             0.0    ...               0.0     0.0       1.0   
1             0.0             0.0    ...               0.0     0.0       1.0   
2             0.0             0.0    ...               0.0     0.0       1.0   
3             0.0             0.0    ...               0.0     1.0       0.0   
4             0.0             0.0    ...               0.0     1.0       0.0   

   type 3_right_y  type 4_y  type 5_y  type 6  type 7_y  type 8_right_y  \
0             0.0       0.0       0.0     0.0       0.0             0.0   
1             0.0       0.0       0.0     0.0       0.0             0.0   
2             0.0       0.0       0.0     0.0       0.0             0.0   
3             0.0       0.0       0.0     0.0       0.0             0.0   
4             0.0       0.0       0.0     0.0       0.0             0.0   

   type 9_y  
0       0.0  
1       0.0  
2       0.0  
3       0.0  
4       0.0  

[5 rows x 314 columns]

#### Step 5 : Reduce dimensions

In [124]:

# transform the vector for training set
X_test_transform = pca.fit_transform(X_test)

# put it into a dataframe
X_test_transform = pd.DataFrame(X_test_transform)
print X_test_transform.head()

          0         1         2         3         4         5         6   \
0  27.568422  3.130501  1.416779  1.275329  0.280165 -0.265865  0.162398   
1  41.573366  2.890950  1.359723 -0.269297  0.554963 -0.662387 -1.043203   
2  41.574411  2.877926  1.272128 -0.459267  0.502563 -0.404015 -0.987178   
3 -46.543884  0.258002 -0.122537 -0.703620 -0.926126  0.470036 -0.106347   
4 -46.543065  0.346750  0.566293 -0.663952 -0.728687  0.009842 -0.256566   

         7         8         9     ...           40        41        42  \
0 -0.902504  0.015296  0.678656    ...     0.055981 -0.036713  0.471439   
1 -0.326068 -0.380162 -0.065214    ...     0.313879 -0.231620 -0.431102   
2 -0.476238 -0.129361 -0.023998    ...     0.255975 -0.571686 -0.007371   
3  0.607468  0.291233 -1.403201    ...     0.243033  0.476894  0.317730   
4  0.462832  0.040748 -1.276175    ...     0.431738  0.751787  0.328953   

         43        44        45        46        47        48        49  
0 -0.258120 -0.226

#### Step 6 : Make Predictions

In [125]:
#Make prediction over the trained RandomForest alg using the testing sample:

y_RandomForest_predict_test = my_RandomForest.predict(X_test_transform)

print(y_RandomForest_predict_test)
print(y_RandomForest_predict_test.shape)

[1 1 1 ..., 1 1 1]
(40092L,)


#### Step 7 : Save the results into a csv file

In [132]:
np.savetxt('C:/Masters/5661/Project REDHAT/myresultRedHat.csv', 
           y_RandomForest_predict_test, 
           delimiter=',', 
           fmt='%3i', 
           header='Results of Random tree prediction')
